consensus/istanbul/backend/announce.go

```diff
@@ -370,52 +369,42 @@ func (sb *Backend) generateAndGossipAnnounce(version uint) error {
 		return err
 	}
 
-	// Add the generated announce message to this node's cache
-	sb.cachedAnnounceMsgsMu.Lock()
-	defer sb.cachedAnnounceMsgsMu.Unlock()
-	sb.cachedAnnounceMsgs[announceAddress] = &announceMsgCachedEntry{MsgVersion: version, MsgPayload: payload}
-
 	return sb.Multicast(nil, payload, istanbulAnnounceMsg)
 }
 
-// This function is a helper function for generateAndGossipAnnounce.
-// It will create the latest announce msg for this node with a given version.
-func (sb *Backend) generateAnnounce(version uint) (*istanbul.Message, common.Address, error) {
+// This function is a helper function for generateAndGossipAnnounce that will
+// use the version for this node's address in the signed announce version db
```

Comment:
Nit: Commenting that this is a helper function isn't particularly helpful to the reader. Could you make this more descriptive.